index.js

```diff
@@ -417,17 +422,23 @@
     format,     // :: (String -> String, String -> String -> String) -> String
     supertypes, // :: Array Type
     test,       // :: Any -> Boolean
-    keys,       // :: Array String
-    types       // :: StrMap { extractor :: a -> Array b, type :: Type }
+    tuples      // :: Array (Array3 String (a -> Array b) Type)
```

Comment:
Currently we take `keys` and `types`, and assume that the elements of `keys` correspond to the keys of `types`. With this change it's not possible to have incoherent data. :)